server/src/main/java/org/elasticsearch/action/search/ExpandSearchPhase.java

```diff
@@ -60,6 +72,9 @@ public void run() {
             MultiSearchRequest multiRequest = new MultiSearchRequest();
             if (collapseBuilder.getMaxConcurrentGroupRequests() > 0) {
                 multiRequest.maxConcurrentSearchRequests(collapseBuilder.getMaxConcurrentGroupRequests());
+            } else if (context.getRequest().preference() != null) {
```

Comment:
I'm not sure how much "preference" contributes to being the problem. If you have the typical setup with number_of_replicas:1, the same data-node level burst of requests is likely to occur even if you are not using preference here.

I'm wondering if we should always bound this in some way when doing internal multi-search requests, or whether we should seek at addressing this at the multi-search request handling level (e.g. some sensible default for maxConcurrentSearchRequests).

Reply:
> I'm not sure how much "preference" contributes to being the problem.

The customer reported this issue has 14 replicas for their target indies.

> I'm wondering if we should always bound this in some way when doing internal multi-search requests

+1

> whether we should seek at addressing this at the multi-search request handling level (e.g. some sensible default for maxConcurrentSearchRequests).

I think the default maxConcurrentSearchRequests in TransportMultiSearchAction is too large https://github.com/elastic/elasticsearch/blob/cce5ad2e4fdfe20fdce0972a1e20a9c4a722a75a/server/src/main/java/org/elasticsearch/action/search/TransportMultiSearchAction.java#L103-L113.